In [1]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All" 
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

**<font color = DarkSlateBlue size = 5>Table of Contents</font>**

* [**<font color = SlateBlue size = 4>Import related libraries</font>**](#section-one)

* [**<font color = SlateBlue size = 4>Read data and Inspect data</font>**](#section-two)

* [**<font color = SlateBlue size = 4>Data Cleaning</font>**](#section-three)

* [**<font color = SlateBlue size = 4>Data Analysis</font>**](#section-four)

* [**<font color = SlateBlue size = 4>Conclusion</font>**](#section-five)

<a id="section-one"></a>
**<font color = DarkSlateBlue size = 5>Import related libraries</font>**

In [2]:
import numpy as np
import pandas as pd
import plotly.express as px

<a id="section-two"></a>
**<font color = DarkSlateBlue size = 5>Read data and Inspect data</font>**

In [3]:
# Read data
df= pd.read_csv('Billionaire.csv')

In [4]:
# Show first five rows of data
df.head()

Name NetWorth        Country         Source  Rank  \
0                Jeff Bezos   $177 B  United States         Amazon     1   
1                 Elon Musk   $151 B  United States  Tesla, SpaceX     2   
2  Bernard Arnault & family   $150 B         France           LVMH     3   
3                Bill Gates   $124 B  United States      Microsoft     4   
4           Mark Zuckerberg    $97 B  United States       Facebook     5   

    Age          Industry  
0  57.0        Technology  
1  49.0        Automotive  
2  72.0  Fashion & Retail  
3  65.0        Technology  
4  36.0        Technology

In [5]:
# Information about data
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2755 entries, 0 to 2754
Data columns (total 7 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   Name      2755 non-null   object 
 1   NetWorth  2755 non-null   object 
 2   Country   2755 non-null   object 
 3   Source    2755 non-null   object 
 4   Rank      2755 non-null   int64  
 5   Age       2676 non-null   float64
 6   Industry  2755 non-null   object 
dtypes: float64(1), int64(1), object(5)
memory usage: 150.8+ KB


<a id="section-three"></a>
**<font color = DarkSlateBlue size = 5>Data cleaning</font>**

In [6]:
df.isnull().sum()

Name         0
NetWorth     0
Country      0
Source       0
Rank         0
Age         79
Industry     0
dtype: int64

In [7]:
df = df.dropna()

In [8]:
df['NetWorth'] = df['NetWorth'].str.strip('$')
df['NetWorth'] = df['NetWorth'].str.strip('B')

In [9]:
df['NetWorth'] = df['NetWorth'].astype(float)
df['Age'] = df['Age'].astype(np.int64)

<a id="section-four"></a>
**<font color = DarkSlateBlue size = 5>Data analysis</font>**

**<font color = SlateBlue size = 4 >-World-</font>**

In [26]:
df1 = df.sort_values(by = ['NetWorth'],ascending = False).reset_index().head(20)

fig = px.bar(df1, x= 'Name', y ='NetWorth', color = 'NetWorth',
             color_continuous_scale = 'sunset',
             hover_data = ['Source','Age'],
             labels= {"NetWorth":"Networth(in Billions)"})

fig.update_xaxes(tickangle=45, tickfont=dict(color='Indigo'))

fig.update_layout(title = 'Top 20 Billionaires in the world',
                  title_font = dict(size = 22, family = 'Balto', color = 'Indigo'),
                  title_x = 0.5)

fig.show()

In [11]:
df1 = df['Source'].value_counts().reset_index().head(20)

fig = px.bar(df1, x = df1['index'], y = df1['Source'], color =df1['Source'],
             color_continuous_scale ='sunset',
             labels = {"index":"source","Source":"count"})

fig.update_xaxes(tickangle=45, tickfont=dict(color='Indigo'))

fig.update_layout(title = 'Top 20 Sources with the most billionaires in the world',
                  title_x = 0.5,
                  title_font = dict(size = 22, family = 'Balto', color = 'Indigo'),
                  xaxis = dict(title = 'Source'),
                  yaxis = dict(title = 'Count')
                 
                 )
fig.show()

In [12]:
df1 = df['Industry'].value_counts().reset_index()

fig = px.pie(df1, 
             values = 'Industry', 
             names = 'index', 
             color_discrete_sequence = px.colors.sequential.Sunset,
             labels = {"index":"Industry","Industry":"Count"})

fig.update_layout(title = '#This pie shows a breakdown of industries of billionaires in the world',
                  title_font = dict(size = 18, family = 'Balto'))

fig.update_traces(textposition = 'inside', textinfo = 'percent+label')
fig.show()

In [13]:
df1 = df['Country'].value_counts().reset_index().head(10)
fig = px.pie(df1,
             values = 'Country', 
             names = 'index', 
             color_discrete_sequence = px.colors.sequential.Rainbow,
             labels = {"index":"Country","Country":"Count"}
            )

fig.update_traces(textposition = 'inside', textinfo = 'percent+label')

fig.update_layout(title = 'Top 10 Countries with the most billionaires',
                  title_x = 0.5,
                  title_font = dict(size = 22, family = 'Balto', color = 'DarkBlue'))
fig.show()

In [ ]:
df1 = df['Country'].value_counts().reset_index().tail(30)

df1 = df1.rename(columns = {"index":"country"})
df1 = df1.rename(columns = {"Country":"count"})

fig = px.bar(df1, y =  df1['country'], x = df1['count'],color =df1['count'], 
             color_continuous_scale = 'teal',
             orientation = 'h')
fig.update_layout(title = 'Countries with the least amount of billionaires',
                  title_x = 0.5,
                  title_font = dict(size = 22, family = 'Balto', color = 'Teal'),
                  xaxis = dict(title = 'Count'),
                  yaxis = dict(title = 'Country'))
fig.show()

**<font color = SlateBlue size = 4 >-United States-</font>**

In [ ]:
dfus = df[df['Country']=='United States']

In [ ]:
df1 = dfus.sort_values(by = ['NetWorth'],ascending = False).reset_index().head(10)

fig = px.bar(df1, x ='Name', y = 'NetWorth', color = 'NetWorth',
             color_continuous_scale = 'rdbu',
             hover_data = ['Source','Age'],
             labels= {"NetWorth":"Networth(in Billions)"}
            )
fig.update_layout(title = 'Top 10 Billionaires in the United States',
                  title_x = 0.5,
                  title_font = dict(size = 22, family = 'Balto',color = 'DarkBlue'))
fig.show()

In [ ]:
df1 = dfus['Source'].value_counts().reset_index().head(10)

fig = px.bar(df1, x = df1['index'], y = df1['Source'], color =df1['Source'],
             color_continuous_scale ='rdbu',
             labels = {"index":"source","Source":"count"})

fig.update_layout(title = 'Top 10 Sources with the most billionaires in the United States',
                  title_x = 0.5,
                  title_font = dict(size = 22, family = 'Balto', color = 'DarkBlue'),
                  xaxis = dict(title = 'Source'),
                  yaxis = dict(title = 'Count'))
fig.show()

In [ ]:
df1 = dfus['Industry'].value_counts().reset_index()

fig = px.pie(df1,
             values = 'Industry', 
             names = 'index', 
             color_discrete_sequence = px.colors.sequential.RdBu,
             labels = {"index":"Industry","Industry":"Count"})

fig.update_layout(title = '#This pie shows a breakdown of industries of billionaires in the United States',
                  title_font = dict(size = 18, family = 'Balto'))

fig.update_traces(textposition = 'inside', textinfo = 'percent+label')
fig.show()

**<font color = SlateBlue size = 4 >-China-</font>**

In [ ]:
dfch = df[df['Country']=='China']

In [20]:
df1 = dfch.sort_values(by = ['NetWorth'],ascending = False).reset_index().head(10)

fig = px.bar(df1, x = 'Name', y = 'NetWorth', color = 'NetWorth', 
             color_continuous_scale = 'reds',
             hover_data = ['Source','Age'],
             labels= {"NetWorth":"Networth(in Billions)"}
            )

fig.update_layout(title = 'Top 10 Billionaires in China',
                  title_x = 0.5,
                  title_font = dict(size = 22, family = 'Balto',color = 'DarkRed'))
fig.show()

In [21]:
df1 = dfch['Source'].value_counts().reset_index().head(10)

fig = px.bar(df1, x = df1['index'], y = df1['Source'], color =df1['Source'],
             color_continuous_scale ='reds',
             labels = {"index":"source","Source":"count"})

fig.update_layout(title = 'Top 10 Sources with the most billionaires in China',
                  title_x = 0.5,
                  title_font = dict(size = 22, family = 'Balto', color = 'DarkRed'),
                  xaxis = dict(title = 'Source'),
                  yaxis = dict(title = 'Count'))
fig.show()

In [22]:
# This pie shows a breakdown of industries of billionaires in China
df1 = dfch['Industry'].value_counts().reset_index()

fig = px.pie(df1,
             values = 'Industry', names = 'index', 
             color_discrete_sequence = px.colors.sequential.Reds,
             labels = {"index":"Industry","Industry":"Count"})

fig.update_layout(title = '#This pie shows a breakdown of industries of billionaires in China',
                  title_font = dict(size = 18, family = 'Balto'))

fig.update_traces(textposition = 'inside', textinfo = 'percent+label')
fig.show()

In [23]:
df1 = df['Country'].value_counts().reset_index()
df1 = df1.rename(columns = {"index":"country"}) 
df1 = df1.rename(columns = {"Country":"count"}) 

fig = px.choropleth(df1, 
                    locations = 'country',
                    locationmode = 'country names',
                    color = 'count',
                    color_continuous_scale = 'twilight')

fig.update_layout(title = 'World - The number of Billionaires',
                  title_x = 0.5,
                  title_font = dict(size = 22, family = 'Balto', color = 'DarkSlateBlue'),
                  geo = dict(showframe = False,
                             showcoastlines = True,
                             projection_type = 'natural earth'))

fig.show()

In [24]:
df1 = df.groupby("Country")['NetWorth'].sum().reset_index()

fig = px.choropleth(df1, 
                    locations = 'Country',
                    locationmode = 'country names',
                    color = 'NetWorth',
                    labels= {"NetWorth":"Networth(in Billions)"},
                    color_continuous_scale = 'rdpu')

fig.update_layout(title = '#This map shows the total networth of billionaires in various countries',
                  title_font = dict(size = 18, family = 'Balto'),
                  geo = dict(showframe = False,
                             showcoastlines = True,
                             projection_type = 'equirectangular'))

fig.show()

**<font color = RebeccaPurple size = 4>Animated map</font>**

In [25]:
df1 = df.groupby("Country")['NetWorth'].sum().sort_values(ascending = False).reset_index().head(20)

fig = px.choropleth(df1, 
                    locations = 'Country',
                    locationmode = 'country names',
                    color = 'NetWorth',
                    animation_frame = 'Country',
                    labels= {"NetWorth":"Networth(in Billions)"},
                    color_continuous_scale = 'plasma')

fig.update_layout(title = '#This map shows top 20 countries by the total networth of billionaires',
                  title_font = dict(size = 18, family = 'Balto'),
                  geo = dict(showframe = False,
                             showcoastlines = True,
                             projection_type = 'equirectangular'))

fig.show()

<a id ="section-five"></a>
**<font color = DarkSlateBlue size = 5>Conclusion</font>**

* **Jeff Bezos and Elon Musk who are from United States are richest in the world and their Networth are more than 150 billion.**
* **Real Estate and Pharmaceuticals are the sources with the most billionaires in the world.**
* **Finance & investment and Technology are the industries with the most billionaires in the world.**
* **Top countries with the most number of billionaires are United Stated and China. And then, over 66 percent of billionaires are from these two countries.**
* **Hedge funds and Real Estate are the sources with the most billionaires in the United States.**
* **Zhong Shanshan and Ma Huateng are richest in China and their Networth are more than 65 billion.**
* **Manufacturing and Technology are the industries with the most billionaires in China.**

**<font color = DarkSlateBlue size = 5>Thank you for reading!</font>**

**Please consider upvoting & checking out more of my work if you found this interesting & valuable! Thanks so much!**

[Link to My Kaggle](http://www.kaggle.com/carriech)